frontend/src/app/models/results.ts

```diff
@@ -10,9 +12,9 @@ import * as _ from 'lodash';
  * for fetching results. Results will be loaded when the query model
  * or the parameters update
  */
-export abstract class Results<Parameters, Result> {
+export abstract class Results<Parameters extends object, Result> extends StoreSync<Parameters> {
```

Comment:
Why does Parameters need to extend object?

Reply:
In this class, that's needed because the `State` type in `StoreSync` must be an object, so you can't allow `any` in the child class.

As for the `StoreSync`, it's needed to allow partial states as input to the `setParams` (i.e. only specify the new value for _some_ properties).